File: webapp/src/lib/repositories/dream-repository.ts
import { getRepository } from 'typeorm';
import { Dream } from '../entities/Dream';

export class DreamRepository {
  private dreamRepository = getRepository(Dream);

    async create(dream: string): Promise<void> {
        // TODO - fix this
        // return this.dreamRepository.create();
    }

    async findAll(): Promise<Dream[]> {
        return this.dreamRepository.find();
    }
}

File: server/repositories/dream_repository.go
package repositories

import (
	"dreams/models"

	"gorm.io/gorm"
)

type DreamRepository struct {
	db *gorm.DB
}

func NewDreamRepository(db *gorm.DB) *DreamRepository {
	return &DreamRepository{db: db}
}

func (r *DreamRepository) Create(content string) error {
	dream := models.Dream{
		Content: content,
	}
	return r.db.Create(&dream).Error
}

func (r *DreamRepository) FindAll() ([]models.Dream, error) {
	var dreams []models.Dream
	err := r.db.Find(&dreams).Error
	return dreams, err
}
